lib/util/Components.js

```diff
@@ -881,7 +881,7 @@ function componentRule(rule, context) {
     },
 
     ArrowFunctionExpression(node) {
-      if (node.async) {
+      if (node.async && node.generator) {
```

Comment:
```suggestion
      if (node.async) {
```
arrows can't be generators, so i think this change will make coverage pass

Reply:
```diff
-     if (node.async && node.generator) {
+     if (node.async) {
```

This would indeed make the coverage pass, but the if statement is an early return to prevent async function to be considered as components. Which is what we want to achieve. I went with removing the early return and it seems to pass all checks!